RELEASING.md

```diff
@@ -0,0 +1,38 @@
+* Choose the right version number:
+  * `x.0.0` is reserved for incompatible changes and require a milestone cycle.
+  * `x.y.0` is reserved for compatible changes.
+  * `x.y.z` is reserved for bugfixes that don't change the public API,
+* Tag the release:
+  * The tag must be called `vx.y.z`, e.g. `v3.0.0`.
+  * `git tag -a vx.y.z -m "vx.y.z"`
+  * `git push upstream --tags`
+  * Do not create a release on GitHub just yet. Creating a release on GitHub
+    sends out a notification to repository watchers, and the release isn't ready
+    for that yet.
+* Wait for [the Travis CI job](https://travis-ci.org/scalameta/metals/branches)
+  in Active Branches to build the binaries and stage them to Sonatype.
+* While waiting for Travis, update the milestones:
+  * https://github.com/scalameta/metals/milestones
+  * Close the milestone or milestones corresponding to the release.
+    For example, for 3.3.0, we closed both 3.2.1 and 3.3.0 (we never
+    released 3.2.1, so all its tickets went straight to 3.3.0).
```

Comment:
are this invented numbers? Otherwise it's 0.3.0 I think.

Reply:
This was copy-pasted from the  scalameta repo and adjusted, I kept the v3.3.0 example but reformulated so it makes sense here.